grpc-gateway/service/deleteResource_test.go

```diff
@@ -123,3 +131,85 @@ func TestRequestHandlerDeleteResource(t *testing.T) {
+	defer cancel()
+
+	tearDown := service.SetUp(ctx, t)
+	defer tearDown()
+	ctx = kitNetGrpc.CtxWithToken(ctx, oauthTest.GetDefaultAccessToken(t))
+
+	conn, err := grpc.Dial(config.GRPC_HOST, grpc.WithTransportCredentials(credentials.NewTLS(&tls.Config{
+		RootCAs: test.GetRootCertificatePool(t),
+	})))
+	require.NoError(t, err)
+	defer func() {
+		_ = conn.Close()
+	}()
+	c := pb.NewGrpcGatewayClient(conn)
+	_, shutdownDevSim := test.OnboardDevSim(ctx, t, c, deviceID, config.GW_HOST, test.GetAllBackendResourceLinks())
+	defer shutdownDevSim()
+	const switchID1 = "1"
+	const switchID2 = "2"
+	test.AddDeviceSwitchResources(ctx, t, deviceID, c, switchID1, switchID2)
+	time.Sleep(200 * time.Millisecond)
+
+	const correlationID = "delete"
+	subClient, err := grpcgwClient.New(c).GrpcGatewayClient().SubscribeToEvents(ctx)
+	require.NoError(t, err)
+	err = subClient.Send(&pb.SubscribeToEvents{
+		CorrelationId: correlationID,
+		Action: &pb.SubscribeToEvents_CreateSubscription_{
+			CreateSubscription: &pb.SubscribeToEvents_CreateSubscription{
+				DeviceIdFilter: []string{deviceID},
+				EventFilter: []pb.SubscribeToEvents_CreateSubscription_Event{
+					pb.SubscribeToEvents_CreateSubscription_RESOURCE_UNPUBLISHED,
+				},
+			},
+		},
+	})
+	require.NoError(t, err)
+	waitForOperationProcessedEvent(t, subClient, correlationID)
+
+	fileWatcher, err := fsnotify.NewWatcher()
+	require.NoError(t, err)
+	defer func() {
+		err := fileWatcher.Close()
+		require.NoError(t, err)
+	}()
+
+	cfg := raTest.MakeConfig(t)
+	raConn, err := client.New(config.MakeGrpcClientConfig(cfg.APIs.GRPC.Addr), fileWatcher, log.Get(), trace.NewNoopTracerProvider())
+	require.NoError(t, err)
+	defer func() {
+		_ = raConn.Close()
+	}()
+	rac := raService.NewResourceAggregateClient(raConn.GRPC())
+
+	respUnpublish, err := rac.UnpublishResourceLinks(ctx, &commands.UnpublishResourceLinksRequest{
+		Hrefs:    []string{test.TestResourceSwitchesInstanceHref(switchID2)},
+		DeviceId: deviceID,
+		CommandMetadata: &commands.CommandMetadata{
+			ConnectionId: uuid.Must(uuid.NewRandom()).String(),
+			Sequence:     0,
+		},
+	})
+	require.NoError(t, err)
+	require.Len(t, respUnpublish.UnpublishedHrefs, 1)
+	require.Equal(t, respUnpublish.UnpublishedHrefs[0], test.TestResourceSwitchesInstanceHref(switchID2))
+	waitForResourceUnpublished(t, subClient, deviceID, correlationID, []string{test.TestResourceSwitchesInstanceHref(switchID2)})
+
+	_, err = c.DeleteResource(ctx, &pb.DeleteResourceRequest{
+		ResourceId: commands.NewResourceID(deviceID, test.TestResourceSwitchesInstanceHref(switchID2)),
+	})
+	require.NoError(t, err)
+
+	_, err = c.DeleteResource(ctx, &pb.DeleteResourceRequest{
+		ResourceId: commands.NewResourceID(deviceID, test.TestResourceSwitchesInstanceHref(switchID1)),
+	})
+	require.NoError(t, err)
+	waitForResourceUnpublished(t, subClient, deviceID, correlationID, []string{test.TestResourceSwitchesInstanceHref(switchID1)})
+}
```

Comment:
getResourceLinks and checik if switchID2 is there <- with current iotivity-lite it fails, with fixed it will pass